torch/csrc/distributed/rpc/py_rref.cpp

```diff
@@ -108,8 +108,10 @@ PyRRef::PyRRef(const py::object& value, const py::object& type_hint)
     : PyRRef([&value, &type_hint]() {
         TypePtr elem_type = tryInferTypeWithTypeHint(value, type_hint);
         auto rref = RRefContext::getInstance().createOwnerRRef(elem_type);
-        py::object copy(value); // increases refcount
-        IValue ivalue = jit::toIValue(std::move(copy), elem_type);
+        // jit::toIValue takes a py::handle as the first argument, and it calls
+        // py::handle.cast<py::object>() to incref of provided value. The
+        // returned ivalue will keep the reference alive.
+        IValue ivalue = jit::toIValue(value, elem_type);
```

Comment:
I think this simplification makes sense.

There is one subtle change here. 

If we have this `py::object` copy, the passed-in `const py::object&` must be kept alive until `py::object copy(value);` to make `PyRRef::PyRRef` work correctly.

If we remove this `py::object` copy, the passed-in `const py::object&` must be kept alive until the `py::handle.cast<py::object>()` in  `jit::toIValue(..)`.

So let's mention this lifetime assumption on the passed-in `const py::object&`, and why we need this the pybind11 binding for this constructor to be holding GIL.

Reply:
comments added